### DISCUSSION TITLE
Opening and editing excel (xlsx) files. Is it possible?

### DISCUSSION THREAD

**Discussion by SebastianMpl** [dathere/qsv#2221] [Q&A] [answered]
Hi,

Is there a way to open and edit xlsx file? I found only that with excel command I can convert the file to csv and for e.g dump it to my terminal. Is there a way to modify the file and save it somehow or the only way is somehow through pipes?

Could someone share a small snippet of example command?

Thank you,

**Comment by ondohotola** [comment]
I would love to have such a feature, but it'll be extremely difficult to save formulas or to avoid messing up a formula...

**Comment by jqnatividad** [comment]
Writing an Excel file is on the backlog - https://github.com/jqnatividad/qsv/issues/1474.

Editing, however, as @ondohotola pointed out, will be a challenge as folks do all kinds of stuff on Excel that will be hard to anticipate.
 
If we do implement editing, changes will be limited to a new sheet, not an existing sheet.

FYI, that's actually what we'll currently developing on qsv pro - calling the qsv pro API directly from Excel to create a new sheet with statistics of the current sheet.

**Reply by rzmk** [reply, marked as answer]
Here's a preview of using an Office Script from the current worksheet with an unreleased version of the qsv pro API to generate stats in a new worksheet in Excel (subject to change):

![excel-qsv-stats-everything-office-script-qsv-pro](https://github.com/user-attachments/assets/8b80c7d4-2992-4c23-a3cf-9c583163dac6)
